src/tests/functional/shared_test_classes/include/shared_test_classes/base/utils/generate_static_shapes.hpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+using StaticShapesMap = std::map<ov::NodeTypeInfo, std::function<ov::Shape(
+        const std::shared_ptr<ov::Node>& node,
+        size_t port,
+        uint64_t dimMin,
```

Comment:
Looks like, we can extract `dimMin`  & `dimMax` from parameter dynamic shape... What do you think about it?
So, we can generate whole 3 static shapes using one function call

Reply:
I added these parameters for the case with fully dynamic shapes (for example `[128, -1, -1]`). When you dont know intervals for dynamic dimensions. Its needed to avoid huge dimensions.

As far as I understand this algorithm is described here: https://github.com/openvinotoolkit/openvino/blob/07fd0f97df514c04d9f4a8284be0a8efbddd9860/src/tests/functional/plugin/conformance/test_runner/op_conformance_runner/src/read_ir/read_ir.cpp#L261-L271

I just moved this logic to new generator.

Let me know please if I should remove these parameters anyway.

Reply:
Correct. I meant that we can the logic inside function to avoid extra parameters in function call because they are the same  and can be limited by char [1, 128) :)

Reply:
I have deleted the extra parameters. I transferred this logic inside inside the genereate-functions (https://github.com/openvinotoolkit/openvino/pull/24125/commits/dec8fc97d918ef0267d1b21c877f53de6e9d1735)